alacritty_terminal/src/term/mod.rs

```diff
@@ -1457,6 +1457,57 @@ impl<T> Term<T> {
             rendered: false,
         }
     }
+
+    fn grid_to_string_between(&self, y0: Line, y1: Line) -> String {
+        let grid = &self.grid;
+
+        let mut s = String::with_capacity(*(y1 - y0) * *grid.cols());
```

Comment:
This will cause problems when y0 is bigger than y1.

Reply:
I'm not sure if it could actually happen.
`grid_to_string_between()` is a private function.
Adding a few assertions might still be a good idea.

Reply:
Yeah, I'd recommend to at the very least add an assertion for something that might cause seriously wrong behavior at runtime.

If it can't happen, it should at least explode with a nice error.

Reply:
I have added them. I don't know if we need any more assertions here.